### PR TITLE
Migrate java-base to use AirbyteProtocol

### PR DESCRIPTION
**Pull request by cgardens**
## What
* migrate java base to use AirbyteMessage as an envelope.
* fix bug where AirbyteProtocolConverter could not handle when json schema sent multiple types (and add test)

With this PR we can run integrations fully using the Airbyte Protocol!!!!! 

If you want to try this out, 1. run the app using the dev configs; 2. build the dev images locally for exchangratesapi source and the csv-destination destination and then 3. update these config files as follows. you should then be able to run exchangerates api to csv destination fully on airbyte protocol. once we have all the other existing sources migrated over, we can officially switch.

`airbyte-config/init/src/main/resources/config/STANDARD_SOURCE/9fed261d-d107-47fd-8c8b-323023db6e20.json`
```
{
  "sourceId": "9fed261d-d107-47fd-8c8b-323023db6e20",
  "name": "exchangeratesapi.io",
  "dockerRepository": "airbyte/source-exchangeratesapi-singer-abprotocol",
  "dockerImageTag": "dev",
  "documentationUrl": "https://hub.docker.com/r/airbyte/integration-singer-exchangeratesapi_io-source"
}
```

`airbyte-config/init/src/main/resources/config/STANDARD_DESTINATION/8be1cf83-fde1-477f-a4ad-318d23c9f3c6.json`
```
{
  "destinationId": "8be1cf83-fde1-477f-a4ad-318d23c9f3c6",
  "name": "Local CSV",
  "dockerRepository": "airbyte/airbyte-csv-destination-abprotocol",
  "dockerImageTag": "dev",
  "documentationUrl": "https://hub.docker.com/r/airbyte/integration-singer-csv-destination"
}
```

### CODE REVIEW

File: airbyte-config/models/src/main/java/io/airbyte/config/AirbyteProtocolConverters.java [hunk 1]
```diff
@@ -77,9 +81,31 @@ public static Schema toSchema(AirbyteCatalog catalog) {
           .withName(airbyteStream.getName())
           .withFields(list.stream().map(item -> new Field()
               .withName(item.getKey())
-              .withDataType(DataType.valueOf(item.getValue().get("type").asText().toUpperCase()))
+              .withDataType(singerTypesToDataType(item.getValue().get("type")))
```
Comment (michel-tricot) [inline airbyte-config/models/src/main/java/io/airbyte/config/AirbyteProtocolConverters.java:84]: Why do we have the word singer in a class that is AB specific?

File: airbyte-config/models/src/main/java/io/airbyte/config/AirbyteProtocolConverters.java [hunk 2]
```diff
@@ -77,9 +81,31 @@ public static Schema toSchema(AirbyteCatalog catalog) {
...
   }
 
+  /**
+   * Singer tends to have 2 types for fields one of which is null. The null is pretty irrelevant, so
+   * look at types and find the first non-null one and use that.
+   *
+   * @param singerTypes - list of types discovered by singer.
+   * @return reduce down to one type which best matches the field's data type
+   */
+  private static DataType singerTypesToDataType(JsonNode node) {
```
Comment (michel-tricot) [inline airbyte-config/models/src/main/java/io/airbyte/config/AirbyteProtocolConverters.java:96]: This is not singer right this is jsonschema

**Review by michel-tricot — APPROVED** [review]

**Review by sherifnada — CHANGES_REQUESTED** [review]
This seems like a hack around that we don't output the correct types from the integration images. Anything ABprotocol-compliant shouldn't output types that aren't specified in ABP imo. 

FWIW I happened to add a fix for this in #610 that makes the integration images output this so this behavior isn't needed in AB core.

File: airbyte-config/models/src/main/java/io/airbyte/config/AirbyteProtocolConverters.java (hunk 2 above)
Comment (cgardens) [reply airbyte-config/models/src/main/java/io/airbyte/config/AirbyteProtocolConverters.java:96]: yes. copy paste error. i stole this function from our other converter, but yes you are right. this is jsonschema type to data type.

File: airbyte-config/models/src/main/java/io/airbyte/config/AirbyteProtocolConverters.java (hunk 1 above)
Comment (cgardens) [reply airbyte-config/models/src/main/java/io/airbyte/config/AirbyteProtocolConverters.java:84]: see comment. purely a name issue. not actually singer specific. will fix tha name. https://github.com/airbytehq/airbyte/pull/609/files#r507447435

**Comment by cgardens** [comment]
> This seems like a hack around that we don't output the correct types from the integration images. Anything ABprotocol-compliant shouldn't output types that aren't specified in ABP imo.

@sherifnada - can you be specific about what you feel is hack? this PR makes sure that the java integrations use the ab protocol and that the worker is able to handle valid json schema. I am not understanding what your concern is.

**Comment by sherifnada** [comment]
@cgardens Maybe the confusion here is around what ABP's type system is. Is it supposed to be all of JsonSchema's types, or whatever is present in `DataType`? I reviewed with the understanding that it's the latter.

**Review by sherifnada — APPROVED** [review]
Nvm - brainfart. Got confused about types.

**Review by jrhizor — APPROVED** [review]

**Pull request merged**